httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/AuthChallengeParser.java

```diff
@@ -58,22 +59,26 @@
     // These private static variables must be treated as immutable and never exposed outside this class
     private static final BitSet TERMINATORS = TokenParser.INIT_BITSET(BLANK, EQUAL_CHAR, COMMA_CHAR);
     private static final BitSet DELIMITER = TokenParser.INIT_BITSET(COMMA_CHAR);
+    private static final BitSet SPACE = TokenParser.INIT_BITSET(BLANK);
 
-    NameValuePair parseTokenOrParameter(final CharSequence buffer, final ParserCursor cursor) {
+    static class ChallengeInt {
 
-        tokenParser.skipWhiteSpace(buffer, cursor);
-        final String token = tokenParser.parseToken(buffer, cursor, TERMINATORS);
-        if (!cursor.atEnd()) {
-            if (buffer.charAt(cursor.getPos()) == BLANK) {
-                tokenParser.skipWhiteSpace(buffer, cursor);
-            }
-            if (!cursor.atEnd() && buffer.charAt(cursor.getPos()) == EQUAL_CHAR) {
-                cursor.updatePos(cursor.getPos() + 1);
-                final String value = tokenParser.parseValue(buffer, cursor, DELIMITER);
-                return new BasicNameValuePair(token, value);
-            }
+        final String schemeName;
+        final List<NameValuePair> params;
+
+        ChallengeInt(final String schemeName) {
+            this.schemeName = schemeName;
+            this.params = new ArrayList<>();
         }
-        return new BasicNameValuePair(token, null);
+
+        @Override
+        public String toString() {
+            return "ChallengeInternal{" +
```

Comment:
I think a `String.format()` would improve readability of this code.

Reply:
@michael-o This may well be true but then we should apply it to `#toString` methods consistently.

Reply:
@ok2c Agreed. Leave it as-is then.